src/generators/ui-schema-gen.ts

```diff
@@ -15,13 +15,16 @@ const createLayout = (layoutType: string): Layout => ({
 /**
  * Derives the type of the jsonSchema element
  */
-const deriveType = (jsonSchema: any): string => {
-    if (jsonSchema.type) {
+const deriveType = (jsonSchema: JsonSchema): string => {
+    if (jsonSchema.type && typeof jsonSchema.type === 'string') {
```

Comment:
Are we only checking for `string` here because of the fall-through cases in `generateUISchema`? If yes, I'd prefer explicit checks for the other primitives as well.